### PR TITLE
feat: token list V2

### PR DESCRIPTION
**Pull request by smartcontracts**
Introducing the one, the only: token list v2! Significantly simplifies
the token listing process by reducing the ways in which users can make
mistakes. Avoids merge conflicts by breaking things out into different
folders. Also introduces CI that gives users automated feedback. What's
not to love?

### CODE REVIEW

**Comment by smartcontracts** [comment]
Such annoying conflicts that it's probably easier if we just force push this to master. I made sure that every token in the current token list is also in the list generated by this PR.

File: src/generate.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            || data.decimals == await contract.decimals(),
+          `${chain} ${folder} decimals mismatch`
+        )
+
+        assert(
+          token.overrides?.symbol !== undefined
+            || data.symbol === await contract.symbol(),
+          `${chain} ${folder} symbol mismatch`
+        )
+
```
Comment (Nicca42) [inline src/generate.js:85]: We should also be checking the addresses match - if you provide an l1 token address you should provide an l2 address unless you are unbridgeable.

File: src/generate.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      }\n`)
+    }
+
+    const logofiles = glob.sync(`${path.join(datadir, folder)}/logo.{png,svg}`)
+    assert(logofiles.length === 1, `must have exactly one logo file (${folder})`)
+
+    for (const [chain, token] of Object.entries(data.tokens)) {
+      console.log(`validating ${folder} on chain ${chain}`)
+
+      if (folder !== 'ETH' && data.nonstandard !== true) {
```
Comment (Nicca42) [inline src/generate.js:67]: We are also not doing any validation of non-standard listings, which we should for sure add soon.

**Review by Nicca42 — APPROVED** [review]

**Review by roninjin10 — APPROVED** [review]
amazing work!

**Pull request merged**